lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -2167,7 +2174,7 @@ def has_different_configuration(self, image):
             cpus=host_config.get('NanoCpus'),
         )
         # Options which don't make sense without their accompanying option
-        if self.parameters.restart_policy:
+        if self.parameters.restart_policy and self.parameters.client.docker_api_version < LooseVersion('1.22'):
```

Comment:
Please move these two lines (without the new part) into the `if` below which updates `config_mapping`.

Reply:
Also done.